### PR TITLE
chore: setup transaction integration tests

### PR DESCRIPTION
**Pull request by fbwoolf**
> Try out this version of the Stacks Wallet - download [extension builds](https://github.com/blockstack/ux/actions/runs/1008934470).

This PR sets up the steps required to test transactions in full page rather than the extension popup (comments are provided). There are a couple sample tests for a contract call as well to get things started. It also removes quite a bit of unused code left behind in the demo page.

cc/ @aulneau @kyranjamie @fbwoolf

### CODE REVIEW

File: src/background/background.ts [hunk 1]
```diff
@@ -44,7 +44,14 @@ chrome.runtime.onConnect.addListener(port => {
           const path = ScreenPaths.GENERATION;
           const urlParams = new URLSearchParams();
           urlParams.set('authRequest', payload);
-          popupCenter({ url: `/popup.html#${path}?${urlParams.toString()}` });
+          // For integration tests, launches in full-page instead of popup
+          if (IS_TEST_ENV) {
+            await chrome.tabs.create({
+              url: chrome.runtime.getURL(`full-page.html#${path}?${urlParams.toString()}`),
+            });
```
Comment (kyranjamie) [inline src/background/background.ts]: I'd suggest a descriptive method here, rather this duplicated comment explain what's going on. 

```ts
if (IS_TEST_ENV) {
  await openAuthRequestInFullPage()
}
```

With the code above everything that's explained in the comment is clear in the code, no comment necessary. 

Perhaps one comment worth noting—that could go inside the body of the helper method—is the _why_, rather than the _what_. Which is, because playwright doesn't support popups properly? Is there a Github issue you could link to in the comment?

Where possible I think we should also avoid using any `chrome.*` api directly, too, which a helper method would also address.

File: test-app/src/components/debugger.tsx [hunk 2]
```diff
@@ -310,7 +311,7 @@ export const Debugger = () => {
 
       <Box>
         <ButtonGroup spacing={4} my="base">
-          <Button mt={3} onClick={callFaker}>
+          <Button data-test={TransactionsSelectors.ContractCall} mt={3} onClick={callFaker}>
```
Comment (kyranjamie) [inline test-app/src/components/debugger.tsx]:
```suggestion
          <Button data-test={TransactionsSelectors.BtnCallContract} mt={3} onClick={callFaker}>
```
Suggested better selector name. This or something more action-based, verby would be clearer.

File: tests/integration/page-objects/demo.page.ts [hunk 3]
```diff
@@ -3,19 +3,7 @@ import { createTestSelector } from '../utils';
...
-  appPriivateKey = '#app-private-key';
-  getStarted = '//span[text()="Get your Secret Key"]';
-  getStartedPopUp = 'css=button>span >> text="Get your Secret Key"';
-  skipBtn = 'text="Skip"';
-  postTextarea = 'div.DraftEditor-root';
-  postBtn = '//span[text()="Post"]';
-  profileBtn = '//*[@title="Your Profile"]';
-  logoutBtn = '//*[contains(text(),"Log out")]';
-  alreadyHaveSecretKeyLink = '//span[text()="Sign in"]';
   openConnectBtn = createTestSelector('sign-up');
```
Comment (kyranjamie) [inline tests/integration/page-objects/demo.page.ts:6]: None of this is used? ☠️   

This is a big clean up, if so 👍🏼

File: test-app/src/components/debugger.tsx [hunk 4]
```diff
@@ -23,6 +23,7 @@ import {
   tupleCV,
   someCV,
 } from '@stacks/transactions';
+import { TransactionsSelectors } from '../../../tests/integration/transactions.selectors';
```
Comment (kyranjamie) [inline test-app/src/components/debugger.tsx]: @ import

**Review by kyranjamie — COMMENTED** [review]
The formatting changes in the `tsconfig.json` aren't how prettier format's JSON.

```
yarn prettier --write tsconfig.json
```

Do you have an editor formatting doing something custom?

File: test-app/src/components/debugger.tsx (hunk 4 above)
Comment (fbwoolf) [reply test-app/src/components/debugger.tsx]: It wouldn't work inside the test-app.

**Comment by fbwoolf** [comment]
> The formatting changes in the `tsconfig.json` aren't how prettier format's JSON.
> 
> ```
> yarn prettier --write tsconfig.json
> ```
> 
> Do you have an editor formatting doing something custom?

Must be from a VSCode update ...seems strange. I'll check.

File: test-app/src/components/debugger.tsx (hunk 4 above)
Comment (aulneau) [reply test-app/src/components/debugger.tsx]: can you try adding the paths? https://github.com/blockstack/stacks-wallet-web/blob/main/test-app/tsconfig.json#L27

**Comment by aulneau** [comment]
> > The formatting changes in the `tsconfig.json` aren't how prettier format's JSON.
> > ```
> > yarn prettier --write tsconfig.json
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Do you have an editor formatting doing something custom?
> 
> Must be from a VSCode update ...seems strange. I'll check.

This is how prettier formats JSON for me...

**Comment by kyranjamie** [comment]
image

**Comment by fbwoolf** [comment]
> 
> This is how prettier formats JSON for me...

Yeah, I actually think it is normal for me too ...should I keep it? Why wouldn't we want it formatted this way?

**Comment by aulneau** [comment]
> > This is how prettier formats JSON for me...
> 
> Yeah, I actually think it is normal for me too ...should I keep it? Why wouldn't we want it formatted this way?

oh this is actually wrong! my editor was set up to use prettier, and so when i did my key combo to format JSON, it did it as your diff shows -- but my file mask for prettier was ignoring JSON (I just added it). It was an editor specific formatting. @fbwoolf wanna confirm yours is not doing the same?

Screen Shot 2021-07-06 at 8 58 29 AM

**Comment by kyranjamie** [comment]
We have Prettier avoid these discussions, so would be best to stick with it. 

Can we change the line in the package json to

```
    "lint:prettier": "prettier --check \"src/**/*.{ts,tsx}\" \"*.{js,json}\"",
```

**Comment by fbwoolf** [comment]
> 
> oh this is actually wrong! my editor was set up to use prettier, and so when i did my key combo to format JSON, it did it as your diff shows -- but my file mask for prettier was ignoring JSON (I just added it). It was an editor specific formatting. @fbwoolf wanna confirm yours is not doing the same?
> 

image

image

I don't have any editor specific config setup for prettier beside formatting on save ...but I did have formatting enabled for JSON.

**Comment by kyranjamie** [comment]
For settings like this, we might as well start using a shared `.vscode/settings.json` file. Would help avoid issues for many new contributors, too.

```
{
  "prettier.documentSelectors": [
    "src/**/*.{ts,tsx}",
    "*.{js,json}"
  ]
}
```

File: test-app/src/components/debugger.tsx (hunk 4 above)
Comment (fbwoolf) [reply test-app/src/components/debugger.tsx]: I can't get the path to be recognized. 🤔

File: tests/integration/page-objects/send-form.selectors.ts [hunk 5]
```diff
@@ -7,5 +7,5 @@ export enum SendFormSelectors {
   InputRecipientField = 'input-recipient-field',
   InputRecipientFieldErrorLabel = 'input-recipient-field-error-label',
 
-  BtnPreviewSendTx = 'bnt-preview-send-tx',
+  BtnPreviewSendTx = 'btn-preview-send-tx',
```
Comment (fbwoolf) [inline tests/integration/page-objects/send-form.selectors.ts:10]: Just caught a typo here.

**Comment by fbwoolf** [comment]
@kyranjamie can you circle back to this in the morning and maybe approve it? I can't seem to get the test paths to be recognized in the test-app ...any ideas? I tried several things but nothing worked.

File: tests/integration/transactions.spec.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  it('validates against insufficient funds when performing a contract call', async () => {
+    const [popup] = await Promise.all([
+      browser.context.waitForEvent('page'),
+      demo.page.click(createTestSelector(TransactionsSelectors.BtnContractCall)),
+    ]);
+    // const [response] = await Promise.all([
+    //   popup.waitForNavigation(),
+    //   popup.click(createTestSelector('account-account-1-0')),
+    // ]);
+    // if (response) console.log(response);
```
Comment (kyranjamie) [inline tests/integration/transactions.spec.ts]: To kill?

**Review by kyranjamie — APPROVED** [review]
Circling back.

Thanks for addressing comments Fara, LGTM

File: tests/integration/transactions.spec.ts (hunk 6 above)
Comment (fbwoolf) [reply tests/integration/transactions.spec.ts]: I need to remove that commented out code, my bad ...but yeah, that is to kill it bc the response should be null. I was just following an example in the documentation.

File: test-app/src/components/debugger.tsx (hunk 4 above)
Comment (aulneau) [reply test-app/src/components/debugger.tsx]: How have you been adding them?

Comment (fbwoolf) [reply test-app/src/components/debugger.tsx]: In the test-app tsconfig with the right route to the test folder (from the test-app src folder), right?

Comment (aulneau) [reply test-app/src/components/debugger.tsx]: This resolves it locally for me:
```diff
{
  "compilerOptions": {
    "target": "es2017",
    "module": "commonjs",
    "lib": ["es2017", "dom"],
    "jsx": "react",
    "sourceMap": true,
    "removeComments": true,
    "noEmit": false,
    "importHelpers": true,
    "downlevelIteration": true,
    "strict": true,
    "noImplicitAny": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "strictBindCallApply": true,
    "strictPropertyInitialization": true,
    "noImplicitThis": true,
    "alwaysStrict": true,
    "skipLibCheck": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "resolveJsonModule": true,
    "moduleResolution": "node",
    "paths": {
      "@store/*": ["./store/*"],
      "@store": ["./store/index"],
      "@dev/*": ["./dev/*"],
      "@components/*": ["./components/*"],
      "@containers/*": ["./containers/*"],
      "@common/*": ["./common/*"],
      "@cards/*": ["./components/cards/*"],
+     "@tests/*": ["../../tests/*"]
    },
    "baseUrl": "./src",
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true
  },
  "include": ["./src/**/*", "./tests/**/*", "./components/**/*", "./common/**/*"]
}
```

Comment (aulneau) [reply test-app/src/components/debugger.tsx]: and in the debugger: 
Screen Shot 2021-07-07 at 11 10 48 AM

Comment (fbwoolf) [reply test-app/src/components/debugger.tsx]: Really?! So weird, definitely tried that exactly. Let me try again, thx for checking it.

Comment (fbwoolf) [reply test-app/src/components/debugger.tsx]: image

Comment (fbwoolf) [reply test-app/src/components/debugger.tsx]: I shutdown VSCode and reopened ...looks like it fixed it. Very annoying! Haha
